configure.ac

```diff
@@ -4161,6 +4160,34 @@ if test "$enable_altsvc" = "yes"; then
   experimental="alt-svc"
 fi
 
+dnl ************************************************************
+dnl check whether ESNI support, if desired, is actualy available
+dnl
+if test "x$want_esni" != "xno"; then
+  AC_MSG_CHECKING([whether ESNI support is available])
+
+  dnl assume NOT and look for sufficient condition
+  ESNI_ENABLED=0
+  ESNI_SUPPORT=''
+
+  dnl OpenSSL with ESNI extension should do the trick
+  if test "x$OPENSSL_ENABLED" == "x1"; then
+    AC_CHECK_HEADERS(openssl/esni.h,
+      ESNI_SUPPORT="OpenSSL with openssl/esni.h"
+      ESNI_ENABLED=1)
```

Comment:
This check ought to actually check that one of the necessary ESNI functions are available in OpenSSL. We've often seen users have a mismatch between headers and library, so just having the header file present is probably not a good enough signal. Something like this:

     AC_CHECK_FUNCS( SSL_ESNI_get_status )

(I don't know what function to actually check for, I just pulled this out of the air)

Reply:
Understood, and agreed.

I expect that this test will be superseded in a later PR which includes functionality. Since this PR is more administrative (feature bit and related build hooks), I thought of leaving the test out, but felt that some example would be useful as a hint.  Otherwise, I might have put

```
  dnl Need a test, but can't be sure what's needed until some implementation is added
  dnl This one will always fail
  if test -z "DUMMY"; then
      ESNI_ENABLED=1

  dnl add 'elif' chain here for per-implementation, implementation-specific tests
  fi
 ```

Reply:
Commit 690b465 checks for the function *SSL_get_esni_status*.